flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -338,6 +339,7 @@ protected FlinkKubeClient createFlinkKubeClient() {
 
 	@Override
 	protected double getCpuCores(Configuration configuration) {
-		return flinkConfig.getDouble(KubernetesConfigOptions.TASK_MANAGER_CPU, numSlotsPerTaskManager);
+		double fallback = configuration.getDouble(KubernetesConfigOptions.TASK_MANAGER_CPU);
```

Comment:
we should also deprecate this option

Reply:
Not sure about this.
We have not expose `TaskManagerOptions#CPU_CORES` to the users. Its annotated `ExcludeFromDocumentation`.
For the same reason, we have not deprecated the cpu config option for Yarn/Mesos.